File: Excel/679 Challenge.py
import pandas as pd

path = "679 Last Item to be Selected.xlsx"
input = pd.read_excel(path, usecols="A:D", skiprows=1, nrows=18)
test = pd.read_excel(path,  usecols="F:G", skiprows=1, nrows=4)

df = input.fillna('')

def last_item(data):
    N = int(data[0])
    items = [i for i in data[1:] if i]
    front = True

    while len(items) > N:
        if front:
            items = items[N:]
        else:
            items = items[:-N]
        front = not front

    return items[-1]

result = pd.DataFrame({
    'Run': df.columns,
    'Last Item': [last_item(df[column].tolist()) for column in df.columns]
})

print(result.equals(test)) # True

File: Excel/600-699/679/679 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/679 Last Item to be Selected.xlsx"
input = read_excel(path, range = "A2:D19")
test  = read_excel(path, range = "F2:G6")
 
last_item <- function(data) {
  N <- as.numeric(data[1])
  items <- na.omit(data[-1])
  front <- TRUE
  
  while (length(items) > N) {
    items <- if (front) items[(N + 1):length(items)] else items[1:(length(items) - N)]
    front <- !front
  }
  
  items[length(items)]
}

result = data.frame(
  Run = names(input),
  items = map_chr(input, last_item)
)

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE